src/lib.rs

```diff
@@ -1,25 +1,14 @@
+pub mod error;
 pub mod index;
 pub mod json_schema;
 pub mod prelude;
 pub mod primitives;
 pub mod regex;
 pub mod vocabulary;
 
-#[cfg(feature = "python-bindings")]
-mod python_bindings;
+use error::Error;
 
-use thiserror::Error;
-
-#[derive(Error, Debug)]
-pub enum Error {
-    #[error("The vocabulary does not allow us to build a sequence that matches the input")]
-    IndexError,
-}
+pub type Result<T, E = Error> = std::result::Result<T, E>;
```

Comment:
Just a suggestion but I think this is related to error handling so it can be in `errors.rs`.

Reply:
It's better there, to be used across all like: `use crate::{Error, Result}`

Reply:
It can be defined in `errors.rs` and re-exported from `lib.rs`. This would ensure `lib.rs` doesn't contain actual logic, rather it'd be just module definitions and re-exports of common items.

But it's not a big deal, feel free to resolve this conversation and keep it in `lib.rs` :+1:

Reply:
Okay, lets move it as you suggest 🤝